File: frontend/src/api/teacher.ts
import request from './request'
import type { WeakKpItem, WrongBookStats } from './learning'

export interface TeacherOverviewSummary {
  student_count: number
  active_students_7d: number
  total_events_7d: number
  wrong_total: number
  wrong_unmastered: number
  mastery_rate: number
}

export interface EventTrendItem {
  date: string
  count: number
}

export interface EnrolledStudentItem {
  user_id: number
  username: string
  joined_at: string
  wrong_count: number
  unmastered_count: number
}

export interface TeacherCourseOverview {
  course: {
    id: number
    name: string
    teacher_id: number
    teacher_name: string
  }
  summary: TeacherOverviewSummary
  students: EnrolledStudentItem[]
  wrong_book_stats: WrongBookStats
  weak_kps: WeakKpItem[]
  event_trend: EventTrendItem[]
}

export function getTeacherCourseOverview(
  courseId: number,
  days = 7,
): Promise<TeacherCourseOverview> {
  return request.get(`/api/v1/teacher/courses/${courseId}/overview`, {
    params: { days },
  })
}

File: backend/app/schemas/teacher.py
from datetime import datetime

from pydantic import BaseModel, Field

from app.schemas.learning import WeakKpItem, WrongBookStatsOut


class CourseBrief(BaseModel):
    id: int
    name: str
    teacher_id: int
    teacher_name: str


class EnrolledStudentItem(BaseModel):
    user_id: int
    username: str
    joined_at: datetime
    wrong_count: int = 0
    unmastered_count: int = 0


class TeacherOverviewSummary(BaseModel):
    student_count: int = 0
    active_students_7d: int = 0
    total_events_7d: int = 0
    wrong_total: int = 0
    wrong_unmastered: int = 0
    mastery_rate: float = 0.0


class EventTrendItem(BaseModel):
    date: str
    count: int


class TeacherCourseOverviewOut(BaseModel):
    course: CourseBrief
    summary: TeacherOverviewSummary
    students: list[EnrolledStudentItem] = Field(default_factory=list)
    wrong_book_stats: WrongBookStatsOut
    weak_kps: list[WeakKpItem] = Field(default_factory=list)
    event_trend: list[EventTrendItem] = Field(default_factory=list)
